src/main/java/org/jboss/pnc/cleaner/archiveservice/ArchivesCleanerImpl.java

```diff
@@ -0,0 +1,172 @@
+                .DELETE()
+                .timeout(Duration.ofSeconds(config.getValue("archive-service.http-client.request-timeout", Long.class)))
+                .header(AUTHORIZATION_STRING, "Bearer " + serviceClient.getAuthToken())
+                .header(CONTENT_TYPE_STRING, "application/json");
+
+        HttpRequest request = builder.build();
+        RetryPolicy<HttpResponse<String>> retryPolicy = new RetryPolicy<HttpResponse<String>>()
+                .withMaxDuration(Duration.ofSeconds(config.getValue("archive-service.running-wait-for", Long.class)))
+                .withMaxRetries(Integer.MAX_VALUE) // retry until maxDuration is reached
+                .withBackoff(
+                        config.getValue("archive-service.running-retry-delay-msec", Long.class),
+                        config.getValue("archive-service.running-retry-max-delay-msec", Long.class),
+                        ChronoUnit.MILLIS)
+                .onSuccess(
+                        ctx -> log
+                                .info("Archival service responded, response status: {}.", ctx.getResult().statusCode()))
+                .onRetry(ctx -> {
+                    String lastError;
+                    if (ctx.getLastFailure() != null) {
+                        lastError = ctx.getLastFailure().getMessage();
+                    } else {
+                        lastError = "";
+                    }
+                    Integer lastStatus;
+                    if (ctx.getLastResult() != null) {
+                        lastStatus = ctx.getLastResult().statusCode();
+                    } else {
+                        lastStatus = null;
+                    }
+                    log.warn(
+                            "Archival service call retry attempt #{}, last error: [{}], last status: [{}].",
+                            ctx.getAttemptCount(),
+                            lastError,
+                            lastStatus);
+                })
+                .onFailure(ctx -> log.error("Unable to call archival service: {}.", ctx.getFailure().getMessage()))
+                .onAbort(e -> log.warn("Archival service call aborted: {}.", e.getFailure().getMessage()));
+
+        log.info("About to call archival service {}.", request.uri());
+        return Failsafe.with(retryPolicy)
+                .with(executor)
+                .getStageAsync(
+                        () -> httpClient.sendAsync(request, HttpResponse.BodyHandlers.ofString())
+                                .thenApply(validateResponse()))
+                .join();
+    }
+
+    private Function<HttpResponse<String>, HttpResponse<String>> validateResponse() {
+        return response -> {
+            if (response.statusCode() >= 200 && response.statusCode() < 300) {
+                return response;
+            } else {
+                throw new FailedResponseException("Response status code: " + response.statusCode());
```

Comment:
Are we interested in message? Can we also log.error it with message?

Reply:
adding, yes might be interesting to know more